### ISSUE TITLE
making kiwix search result  screenreader accessible 

### ISSUE THREAD

**Issue by kaveinthran** [kiwix/kiwix-js#778]
Using nvda from https://www.nvaccess.org/download/
1. with the screen reader loaded, load the kiwix extension and any zim file
2. press tab until you find the search edit box, and input any query
 3. expected result: screen reader should announce results and arrowing up and down should read each result and pressing enter on each result should activate them.
4. actual result: screen reader stay silent and tab and shift+tab did not  reach the results i.e. the results are not in tab order.

**Comment by Jaifroid** [comment]
@kaveinthran Do you know how to fix this?

**Comment by mossroy** [comment]
It's true that we did not work on accessibility of our app for blind people or partially-sighted users.
First step is probably to make tab key browse the search results. Maybe playing with `tabIndex` attribute would do what we need. So that, when test results are displayed, using tab in the search bar does not put the focus on the next button

**Comment by MuhammadBilalBinAtique** [comment]
I think so every app should be screen-reader accessible. Should I work to make search results accessible for screen readers?

**Comment by Jaifroid** [comment]
@MuhammadBilalBinAtique Do you have experience with screen readers, and also with HTML/JS development? If so, we welcome help, and please read [CONTRIBUTING.md](https://github.com/kiwix/kiwix-js/blob/master/CONTRIBUTING.md).

**Comment by MuhammadBilalBinAtique** [comment]
Yes, I have vast experience with HTML/JS development, but not with much with screen readers. If you feel ok, then you can assign it to me.

**Comment by Jaifroid** [comment]
Thank you! That's fine -- thanks for the offer. I've assigned you.

**Comment by MuhammadBilalBinAtique** [comment]
> Thank you! That's fine -- thanks for the offer. I've assigned you.


Thanks

**Comment by devhenode** [comment]
Hello! I’m interested in working on this issue! I have experience with JavaScript and accessibility improvements. Can I take this up?

**Comment by Jaifroid** [comment]
@devhenode Thanks for the offer, but you should await the outcome of #1313 (very similar) since it is currently encountering difficulties passing tests.

**Comment by devhenode** [comment]
Oh okay, I'm presently checking it.

**Comment by devhenode** [comment]
I have read through his approach of fixing the same similar issue, but I enhanced the solution which is to only address labeling the search input properly so that screen readers can recognize and announce it correctly. 

But my initial solution for picking it up was to actually address **_labeling the search input properly_** and **enable the search results themselves readable, navigable, or interactable.** which I Built and tested in my environments and it works

Code Descriptions attached in an image below 

![Image](https://github.com/user-attachments/assets/487cab90-4511-49dd-b949-baa9ebce987e)
![Image](https://github.com/user-attachments/assets/9fc46309-fcca-4022-8bf8-36f8454586fc)

**Comment by Jaifroid** [comment]
Yes of course, so long as it's doing something different from that PR, which it sounds like it definitely is. Please go ahead and make a PR so we can run the tests.

**Comment by Jaifroid** [comment]
Ah, I see you already did. Please link that PR to this issue.

**Comment by devhenode** [comment]
Okay here's the PR [https://github.com/kiwix/kiwix-js/pull/1321](url)

Thanks for the approval
